File: system-test/dotnet/SystemTests/Legacy/Mod05/Base/BaseDriverTest.cs
using SystemTests.TestInfrastructure.Configuration;
using Dsl.Core;
using Driver.Port.External.Erp;
using Driver.Port.MyShop;
using Driver.Adapter.External.Erp;
using Driver.Adapter.MyShop.Api;
using Driver.Adapter.MyShop.Ui;
using Driver.Adapter.External.Tax;
using Xunit;

namespace SystemTests.Legacy.Mod05.Base;

public abstract class BaseDriverTest : BaseConfigurableTest, IAsyncLifetime
{
    protected readonly Dsl.Core.Configuration _configuration;
    protected IMyShopDriver? _shopDriver;
    protected ErpRealDriver? _erpDriver;
    protected TaxRealDriver? _taxDriver;

    protected BaseDriverTest()
    {
        _configuration = LoadConfiguration();
    }

    public virtual Task InitializeAsync()
    {
        return Task.CompletedTask;
    }

    protected void SetUpMyShopApiDriver()
    {
        _shopDriver = new MyShopApiDriver(_configuration.MyShopApiBaseUrl);
    }

    protected async Task SetUpMyShopUiDriverAsync()
    {
        _shopDriver = await MyShopUiDriver.CreateAsync(_configuration.MyShopUiBaseUrl);
    }

    protected void SetUpExternalDrivers()
    {
        _erpDriver = new ErpRealDriver(_configuration.ErpBaseUrl);
        _taxDriver = new TaxRealDriver(_configuration.TaxBaseUrl);
    }

    public virtual async Task DisposeAsync()
    {
        if (_shopDriver != null)
            await _shopDriver.DisposeAsync();
        _erpDriver?.Dispose();
        _taxDriver?.Dispose();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod05/base/BaseDriverTest.java
package com.optivem.shop.systemtest.legacy.mod05.base;

import com.optivem.shop.systemtest.configuration.BaseConfigurableTest;
import com.optivem.shop.systemtest.configuration.Configuration;
import com.optivem.shop.dsl.driver.adapter.external.erp.ErpRealDriver;
import com.optivem.shop.dsl.driver.adapter.external.tax.TaxRealDriver;
import com.optivem.shop.dsl.driver.adapter.shop.api.ShopApiDriver;
import com.optivem.shop.dsl.driver.port.shop.ShopDriver;
import com.optivem.shop.dsl.driver.adapter.shop.ui.ShopUiDriver;
import com.optivem.shop.systemtest.infrastructure.playwright.BrowserLifecycleExtension;
import com.optivem.shop.dsl.common.Closer;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;

public class BaseDriverTest extends BaseConfigurableTest {
    protected Configuration configuration;

    protected ShopDriver shopDriver;
    protected ErpRealDriver erpDriver;
    protected TaxRealDriver taxDriver;

    @BeforeEach
    protected void setUpConfiguration() {
        configuration = loadConfiguration();
    }

    protected void setUpShopUiDriver() {
        shopDriver = new ShopUiDriver(configuration.getShopUiBaseUrl(), BrowserLifecycleExtension.getBrowser());
    }

    protected void setUpShopApiDriver() {
        shopDriver = new ShopApiDriver(configuration.getShopApiBaseUrl());
    }

    protected void setUpExternalDrivers() {
        erpDriver = new ErpRealDriver(configuration.getErpBaseUrl());
        taxDriver = new TaxRealDriver(configuration.getTaxBaseUrl());
    }

    @AfterEach
    void tearDown() {
        Closer.close(shopDriver);
        Closer.close(erpDriver);
        Closer.close(taxDriver);
    }
}
